detectors/gcp/go.mod

```diff
@@ -1,17 +1,19 @@
 module github.com/GoogleCloudPlatform/opentelemetry-operations-go/detectors/gcp
 
-go 1.21
+go 1.22
```

Comment:
Is upgrading the minimum go version intentional ?

Reply:
It is ok.  I just haven't gotten around to doing it yet.  Let me do that separately